api/db/changelog/db.changelog-44-user-recent-resource.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+    <changeSet author="nehasaxena" id="db.changelog-44-cacheTable">
+        <createTable tableName="user_recent_resource">
+            <column name="id" type="bigint" autoIncrement="true">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="user_id" type="bigint">
+                <constraints nullable="false"/>
+            </column>
+            <column name="workspace_id" type="bigint">
+                <constraints nullable="false"/>
+            </column>
+            <column name="cohort_id" type="bigint">
+                <constraints nullable="true"/>
+            </column>
+            <column name="notebook_name" type="varchar(255)">
```

Comment:
The 255 is arbitrary, right? I'd probably bump this to 1024, which is the maximum GCS object name length: https://cloud.google.com/storage/docs/naming#objectnames

If we get a notebook name that long, I'm sure we'll have other problems, but 1024 at least guarantees this SQL won't be the limiting factor.